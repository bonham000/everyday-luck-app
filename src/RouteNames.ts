export enum ROUTE_NAMES {
  HOME = "HOME",
  VIEW_NOTE = "VIEW_NOTE",
  CREATE_NOTE = "CREATE_NOTE",
}
