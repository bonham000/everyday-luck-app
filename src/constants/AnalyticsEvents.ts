/** ========================================================================
 * App events for tracking
 * =========================================================================
 */

enum EVENTS {
  COMPLETED_LESSON = "COMPLETED_LESSON",
  SET_APP_DIFFICULTY = "SET_APP_DIFFICULTY",
  SET_APP_LANGUAGE = "SET_APP_LANGUAGE",
  TOGGLE_AUDIO_SETTING = "TOGGLE_AUDIO_SETTING",
  TOGGLE_AUTO_PROCEED_QUIZ = "TOGGLE_AUTO_PROCEED_QUIZ",
}

/** ========================================================================
 * Export
 * =========================================================================
 */

export default EVENTS;
