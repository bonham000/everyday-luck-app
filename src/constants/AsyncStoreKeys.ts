/** ========================================================================
 * Async Store Keys
 * =========================================================================
 */
export const ASYNC_STORE_KEYS = {
  USER_KEY: "USER_KEY",
  WORD_STUDY_LIST: "WORD_STUDY_LIST",
  CUSTOM_WORD_STUDY_LIST: "CUSTOM_WORD_STUDY_LIST",
  DOCUMENTS_AGREEMENT: "DOCUMENTS_AGREEMENT",
  OFFLINE_UPDATES_FLAG: "OFFLINE_UPDATES_FLAG",
};
