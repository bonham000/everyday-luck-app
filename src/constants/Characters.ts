export const CHINESE_NUMBER_MAP: ReadonlyArray<any> = [
  "另",
  "一",
  "二",
  "三",
  "四",
  "五",
  "六",
  "七",
  "八",
  "九",
  "十",
  "十一",
  "十二",
  "十三",
  "十四",
  "十五",
  "十六",
  "十七",
  "十八",
  "十九",
  "二十",
  "二十一",
];
