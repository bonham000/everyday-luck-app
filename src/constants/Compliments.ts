/** ========================================================================
 * Adjective options for when users pass/fail questions.
 * =========================================================================
 */
export const COMPLIMENTS: ReadonlyArray<string> = [
  "Amazing",
  "Awesome",
  "Excellent",
  "Fabulous",
  "Fantastic",
  "Incredible",
  "Outstanding",
  "Perfect",
  "Remarkable",
  "Spectacular",
  "Splendid",
  "Stellar",
  "Stupendous",
  "Super",
  "Unbelievable",
];

export const ENCOURAGEMENTS: ReadonlyArray<string> = [
  "No good",
  "Nope",
  "Too bad",
  "So close",
  "Almost",
  "One more time",
  "Nearly",
  "Try again",
];
