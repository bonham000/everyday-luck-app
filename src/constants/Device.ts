import { Dimensions } from "react-native";

/** ========================================================================
 * Dimension adjustments for smaller screens
 * =========================================================================
 */

const HEIGHT = Dimensions.get("window").height;

const SMALL_DEVICE = HEIGHT < 725;

/** ========================================================================
 * Export
 * =========================================================================
 */

export { SMALL_DEVICE };
