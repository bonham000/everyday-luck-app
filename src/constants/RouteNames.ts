/** ========================================================================
 * Route names
 * =========================================================================
 */
export enum ROUTE_NAMES {
  APP = "APP",
  HOME = "HOME",
  LIST_SUMMARY = "LIST_SUMMARY",
  LESSON_SUMMARY = "LESSON_SUMMARY",
  QUIZ = "QUIZ",
  DAILY_CHALLENGE = "DAILY_CHALLENGE",
  HSK_TEST_OUT = "HSK_TEST_OUT",
  MULTIPLE_CHOICE_MANDARIN = "MULTIPLE_CHOICE_MANDARIN",
  MULTIPLE_CHOICE_ENGLISH = "MULTIPLE_CHOICE_ENGLISH",
  MULTIPLE_CHOICE_VOICE = "MULTIPLE_CHOICE_VOICE",
  FLASHCARDS = "FLASHCARDS",
  VIEW_ALL = "VIEW_ALL",
  TRANSLATION = "TRANSLATION",
  WELCOME = "WELCOME",
  INTRO = "INTRO",
  ABOUT = "ABOUT",
  CONTACT = "CONTACT",
  SETTINGS = "SETTINGS",
}
