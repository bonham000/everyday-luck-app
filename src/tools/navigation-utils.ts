import {
  DrawerLockMode,
  NavigationActions,
  StackActions,
} from "react-navigation";

import { ROUTE_NAMES } from "@src/constants/RouteNames";

/**
 * Reset the navigation stack to the given route name.
 *
 * @param routeName to reset navigation to
 * @returns navigation reset action
 */
export const resetNavigation = (routeName: ROUTE_NAMES) => {
  return StackActions.reset({
    index: 0,
    actions: [NavigationActions.navigate({ routeName })],
  });
};

const LOCKED = "locked-closed";
const UNLOCKED = "unlocked";

/**
 * Helper to get the locked state for the side menu drawer.
 *
 * @param navigation
 * @returns docker lock state
 */
export const getDrawerLockedState = (navigation: any): DrawerLockMode => {
  const isOnSignIn = isOnSignInScreen(navigation.state);
  const drawerLockMode = isOnSignIn
    ? LOCKED
    : navigation.state.index > 0
    ? LOCKED
    : UNLOCKED;
  return drawerLockMode;
};

/**
 * Helper to determine if the user is on the SignIn screen.
 *
 * @param navigationState
 * @returns true if user is on the SignIn screen.
 */
const isOnSignInScreen = (navigationState: any): boolean => {
  try {
    if (navigationState.routes[0].routeName === ROUTE_NAMES.SIGNIN) {
      return true;
    } else {
      return false;
    }
  } catch (err) {
    return false;
  }
};
