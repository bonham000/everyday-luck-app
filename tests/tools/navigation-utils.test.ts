import { ROUTE_NAMES } from "@src/constants/RouteNames";
import { resetNavigation } from "@src/tools/navigation-utils";

describe("navigation-utils", () => {
  test("resetNavigation", () => {
    const result = resetNavigation(ROUTE_NAMES.HOME);
    expect(result).toMatchSnapshot();
  });

  test.skip("getDrawerLockedState", () => {
    /**
     * TODO: Add test.
     */
    expect(true).toBeTruthy();
  });
});
