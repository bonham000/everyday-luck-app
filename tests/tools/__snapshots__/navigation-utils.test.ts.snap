// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation-utils resetNavigation 1`] = `
Object {
  "actions": Array [
    Object {
      "routeName": "HOME",
      "type": "Navigation/NAVIGATE",
    },
  ],
  "index": 0,
  "key": null,
  "type": "Navigation/RESET",
}
`;
